scheduler/src/cook/mesos/api.clj

```diff
@@ -659,6 +660,73 @@
                      :body "Internal server error."})) )))
 
 
+(defn get-resources
+  "Tells the user what resource types are available in the system, if
+  the current user is an admin."
+  [conn request]
+  (let [user (:authorization/user request)
+        authorized? (is-authorized? user :access cook.authorization/system)]
+    (if authorized?
+      {:status 200
+       :headers {"Content-Type" "application/json"}
+       :body (cheshire/encode (util/get-all-resource-types (d/db conn)))
+       }
+      {:status 403
+       :body "Forbidden"
+       })))
+
+
+(defn get-user-resources
+  "Tells the user what resource shares are in place for the given username, if
+  the current user is an admin."
+  [conn request username]
+  (let [user (:authorization/user request)
+        authorized? (is-authorized? user :access cook.authorization/system)]
+    (if authorized?
+      {:status 200
+       :headers {"Content-Type" "application/json"}
+       :body (cheshire/encode (share/get-share (d/db conn) username))
+       }
+      {:status 403
+       :body "Forbidden"
+       })))
+
+
+(defn set-user-resources!
```

Comment:
There is enough logic in here that I think it is worth using liberator instead of just a function. This will also allow having a single resource for both the get and post of shares.

You can take a look at https://github.com/pjlegato/Cook/pull/2/files#diff-6611b39b65ff49438a4d940b51c85cc0R424 for how we've done it before.